### PR TITLE
[RLLib] Test with CNNs in RLModule tests

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

This PR integrates CNNs into RLModule tests.

Needs https://github.com/ray-project/ray/pull/32069

### CODE REVIEW

**Comment by ArturNiederfahrenhorst** [comment]
The linter errors is not related (this PR has nothing to do with said file).
I assume that it was broken on master to begin with.

File: rllib/algorithms/ppo/tests/test_ppo_rl_module.py [hunk 1]
```diff
@@ -177,18 +188,26 @@ def test_forward_train(self):
...
                 continue
             print(f"[FW={fw} | [ENV={env_name}] | LSTM={lstm}")
-            env = gym.make(env_name)
+            # TODO(Artur): Figure out why this is needed and fix it.
+            if self.env.startswith("ALE/"):
+                env = gym.make("GymV26Environment-v0", env_id=self.env)
+            else:
+                env = gym.make(env_name)
 
+            preprocessor = get_preprocessor(env.observation_space)(
```
Comment (kouroshHakha) [inline rllib/algorithms/ppo/tests/test_ppo_rl_module.py:197]: why do we construct and transform on the same input?

File: rllib/algorithms/ppo/tests/test_ppo_with_rl_module.py [hunk 2]
```diff
@@ -115,8 +115,11 @@ def test_ppo_compilation_and_schedule_mixins(self):
         for fw in framework_iterator(
             config, frameworks=("torch", "tf2"), with_eager_tracing=True
         ):
-            # TODO (Kourosh) Bring back "FrozenLake-v1" and "MsPacmanNoFrameskip-v4"
-            for env in ["CartPole-v1", "Pendulum-v1"]:
+            # TODO (Kourosh) Bring back "FrozenLake-v1"
+            for env in ["CartPole-v1", "Pendulum-v1", "ALE/MsPacmanNoFrameskip-v4"]:
+                if env == "ALE/MsPacmanNoFrameskip-v4" and fw == "tf2":
+                    # CNN not implement in TF2 yet
```
Comment (kouroshHakha) [inline rllib/algorithms/ppo/tests/test_ppo_with_rl_module.py:121]: TODO (Artur)

**Review by kouroshHakha — APPROVED** [review]

File: rllib/algorithms/ppo/tests/test_ppo_rl_module.py (hunk 1 above)
Comment (ArturNiederfahrenhorst) [reply rllib/algorithms/ppo/tests/test_ppo_rl_module.py:197]: We don't transform. `get_preprocessor` returns a type, which takes the observation space as an input (again).

File: rllib/algorithms/ppo/tests/test_ppo_with_rl_module.py (hunk 2 above)
Comment (ArturNiederfahrenhorst) [reply rllib/algorithms/ppo/tests/test_ppo_with_rl_module.py]: Thanks, made the change.

File: rllib/algorithms/ppo/tests/test_ppo_rl_module.py (hunk 1 above)
Comment (kouroshHakha) [reply rllib/algorithms/ppo/tests/test_ppo_rl_module.py:197]: ok got it.Let's do it in two parts to be more clear.
```
preprocessor_cls = get_preprocessor(env.observation_space)
preprocessor = prerprocessor_cls(env.observation_space)
```

Comment (ArturNiederfahrenhorst) [reply rllib/algorithms/ppo/tests/test_ppo_rl_module.py]: Done

File: rllib/core/models/torch/encoder.py [hunk 3]
```diff
@@ -189,7 +190,7 @@ def get_initial_state(self):
 
     @override(Model)
     def _forward(self, inputs: NestedDict, **kwargs) -> NestedDict:
-        x = inputs[SampleBatch.OBS]
+        x = inputs[SampleBatch.OBS].float()
```
Comment (gjoliver) [inline rllib/core/models/torch/encoder.py:193]: this doesn't feel right.
weights can be float16, while input OBS may be int32s ...

Comment (ArturNiederfahrenhorst) [reply rllib/core/models/torch/encoder.py:193]: I'd agree, but we did this in VisionNet forever now. Does torch barf if you choose lower resolution weights?

Comment (ArturNiederfahrenhorst) [reply rllib/core/models/torch/encoder.py:193]: We can convert with something like 
`x = inputs[SampleBatch.OBS].type(self.net.input_dtype)`.

Comment (ArturNiederfahrenhorst) [reply rllib/core/models/torch/encoder.py:193]: Moved the conversions into the Model objects that also create the layers. These layers are float32 and not configurable, but if anyone would overwrite that code, they'd be able to change the precision there.

**Comment by ArturNiederfahrenhorst** [comment]
@gjoliver Running tests again because of the weird lint error that had nothing to do with the files I changed.

**Comment by ArturNiederfahrenhorst** [comment]
After that I believe we can merge.

**Comment by kouroshHakha** [comment]
one of the failing test seems relevant.

**Pull request merged**